src/WebJobs.Script/Http/RetryProxyHandler.cs

```diff
@@ -30,20 +32,37 @@ public RetryProxyHandler(HttpMessageHandler innerHandler, ILogger logger)
 
         protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
         {
+            TaskCompletionSource<ScriptInvocationResult> resultSource = null;
+            if (request.Options.TryGetValue(ScriptConstants.HttpProxyScriptInvocationContext, out ScriptInvocationContext scriptInvocationContext))
+            {
+                resultSource = scriptInvocationContext.ResultSource;
+            }
+
             var currentDelay = InitialDelay;
             for (int attemptCount = 1; attemptCount <= MaxRetries; attemptCount++)
             {
                 try
                 {
+                    if (resultSource is not null && (resultSource.Task.IsFaulted || resultSource.Task.IsCanceled))
+                    {
+                        ExceptionDispatchInfo.Capture(resultSource.Task.Exception).Throw();
+                    }
+
                     return await base.SendAsync(request, cancellationToken);
                 }
                 catch (TaskCanceledException) when (cancellationToken.IsCancellationRequested)
                 {
                     _logger.LogDebug("Request was canceled. Stopping retries.");
                     throw new OperationCanceledException(cancellationToken);
                 }
-                catch (HttpRequestException) when (attemptCount < MaxRetries)
+                catch (HttpRequestException exception) when (attemptCount < MaxRetries)
                 {
+                    if (resultSource is not null && (resultSource.Task.IsFaulted || resultSource.Task.IsCanceled))
```

Comment:
If the reason for the `resultSource` faulting is not an `HttpRequestException`, this check here can be removed so we just wait until the next retry to check/handle `resultSource` faulting. 

I don't know if there's any cases where the reason for failure is `HttpRequestException`, but if there is, we probably want to add a stop here too.

Also, had a placeholder here in line 63, but unsure which error we want to propagate here (`HttpRequestException` or the function invocation `resultSource` exception.

Reply:
If:

```
  if (resultSource is not null && (resultSource.Task.IsFaulted || resultSource.Task.IsCanceled))
  {
      ExceptionDispatchInfo.Capture(resultSource.Task.Exception).Throw();
  }
```

throws an `HttpRequestException`, and this block is entered, there needs to be another check or a retry will be kicked off. Not sure how often this actually happens, but just in case.